common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -643,17 +637,58 @@ def get_problem_html(self, encapsulate=True):
         demand_hints = self.lcp.tree.xpath("//problem/demandhint/hint")
         demand_hint_possible = len(demand_hints) > 0
 
+        # Get the current problem status and generate the answer notification and message.
+        answer_notification_message = None
+        answer_notification_type = None
+        progress = self.get_progress()
+
+        if progress is not None:
```

Comment:
"graded" is a metadata field on any xblock. Normally it would be inherited from the section or subsection, and you should be able to set it there using the XBlockFixtureDesc.

Here is an example where metadata is added in a XBlockFixtureDesc:

````
XBlockFixtureDesc('sequential', 'Locked Subsection', metadata={'visible_to_staff_only': True}).add_children(
                    XBlockFixtureDesc('vertical', 'Test Unit').add_children(
                        XBlockFixtureDesc(
                            'html', 'Html Child in locked subsection', data="<html>Visible only to staff</html>"
                        )
                    )
                )